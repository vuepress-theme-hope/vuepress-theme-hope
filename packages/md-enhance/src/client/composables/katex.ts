import { onMounted } from "vue";

export const useKatexCopy = (): void => {
  onMounted(
    () =>
      import(
        /* webpackChunkName: "katex" */ "katex/dist/contrib/copy-tex.min.js"
      ),
  );
};
