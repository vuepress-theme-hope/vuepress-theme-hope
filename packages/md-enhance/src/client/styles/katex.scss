@import "katex/dist/katex.min.css";

// katex fix
.katex {
  font-size: 1.05em;
}

.katex-display {
  overflow: auto hidden;
  -webkit-overflow-scrolling: touch;
  padding-top: 0.2em;
  padding-bottom: 0.2em;

  &::-webkit-scrollbar {
    height: 3px;
  }

  .katex {
    font-size: 1.21em;
  }
}

.katex-error {
  color: #f00;
}
