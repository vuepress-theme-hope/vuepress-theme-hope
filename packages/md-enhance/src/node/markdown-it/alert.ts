import { alert as _alert } from "@mdit/plugin-alert";
import type { ExactLocaleConfig } from "@vuepress/helper";
import { ensureLeadingSlash } from "@vuepress/helper";
import type { PluginWithOptions } from "markdown-it";
import type { MarkdownEnv } from "vuepress/markdown";
import { resolveLocalePath } from "vuepress/shared";

import type { MarkdownAlertLocaleData } from "../typings/index.js";

export type MarkdownItAlertOptions = ExactLocaleConfig<MarkdownAlertLocaleData>;

export const alert: PluginWithOptions<MarkdownItAlertOptions> = (
  md,
  options = {},
) => {
  md.use(_alert, {
    alertNames: ["important", "note", "tip", "warning", "caution", "info"],
    openRender: (tokens, index): string =>
      `<div class="hint-container ${tokens[index].markup}">\n`,
    titleRender: (tokens, index, _options, env: MarkdownEnv) => {
      const type = tokens[index].markup;
      const relativePath = ensureLeadingSlash(env.filePathRelative ?? "");
      const localePath = resolveLocalePath(options, relativePath);

      return `\
<p class="hint-container-title">${
        options[localePath]?.[type as keyof MarkdownAlertLocaleData] || type
      }</p>
`;
    },
    closeRender: () => "</div>\n",
    deep: true,
  });
};
