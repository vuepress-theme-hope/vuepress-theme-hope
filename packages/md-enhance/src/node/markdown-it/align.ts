import { container } from "./container";

import type { PluginSimple } from "markdown-it";

export const align: PluginSimple = (md) => {
  ["left", "center", "right", "justify"].forEach((name) =>
    md.use((md) =>
      container(md, {
        name,
        openRender: () => `<div style="text-align: ${name}">`,
      })
    )
  );
};
