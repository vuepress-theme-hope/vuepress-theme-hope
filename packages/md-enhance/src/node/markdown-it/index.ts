export * from "./align.js";
export { attrs } from "./attrs.js";
export * from "./chart.js";
export * from "./codeDemo.js";
export * from "./codeTabs.js";
export * from "./container.js";
export * from "./echarts.js";
export * from "./flowchart.js";
export * from "./footnote.js";
export * from "./hint.js";
export * from "./imageMark.js";
export * from "./imageSize.js";
export * from "./katex.js";
export * from "./mathjax.js";
export * from "./lazyLoad.js";
export * from "./mark.js";
export * from "./include.js";
export * from "./mermaid.js";
export * from "./playground/index.js";
export * from "./presentation.js";
export * from "./sub.js";
export * from "./sup.js";
export * from "./tabs.js";
export * from "./tasklist.js";
export * from "./tex.js";
export * from "./vPre.js";
export * from "./stylize.js";
export * from "./uml.js";
export * from "./vuePlayground.js";
