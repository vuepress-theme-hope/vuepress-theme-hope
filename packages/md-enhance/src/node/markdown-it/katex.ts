/**
 * Forked from https://github.com/waylonflinn/markdown-it-katex/blob/master/index.js
 *
 * The MIT License (MIT)
 *
 * Copyright (c) 2016 Waylon Flinn
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

import Katex from "katex";
import { escapeHtml } from "./utils";

import type { default as StateInline } from "markdown-it/lib/rules_inline/state_inline";
import type { PluginWithOptions } from "markdown-it";
import type { RuleInline } from "markdown-it/lib/parser_inline";
import type { RuleBlock } from "markdown-it/lib/parser_block";
import type { KatexOptions } from "katex";

/*
 * Test if potential opening or closing delimieter
 * Assumes that there is a "$" at state.src[pos]
 */
const isValidDelim = (
  state: StateInline,
  pos: number
): { canOpen: boolean; canClose: boolean } => {
  const prevChar = pos > 0 ? state.src.charAt(pos - 1) : "";
  const nextChar = pos + 1 <= state.posMax ? state.src.charAt(pos + 1) : "";

  return {
    canOpen: nextChar !== " " && nextChar !== "\t",

    /*
     * Check non-whitespace conditions for opening and closing, and
     * check that closing delimeter isnâ€™t followed by a number
     */
    canClose: !(
      prevChar === " " ||
      prevChar === "\t" ||
      /[0-9]/u.exec(nextChar)
    ),
  };
};

const inlineTex: RuleInline = (state, silent) => {
  let match;
  let pos;
  let res;
  let token;

  if (state.src[state.pos] !== "$") return false;

  res = isValidDelim(state, state.pos);
  if (!res.canOpen) {
    if (!silent) state.pending += "$";

    state.pos += 1;

    return true;
  }

  /*
   * First check for and bypass all properly escaped delimieters
   * This loop will assume that the first leading backtick can not
   * be the first character in state.src, which is known since
   * we have found an opening delimieter already.
   */
  const start = state.pos + 1;

  match = start;
  while ((match = state.src.indexOf("$", match)) !== -1) {
    /*
     * Found potential $, look for escapes, pos will point to
     * first non escape when complete
     */
    pos = match - 1;
    while (state.src[pos] === "\\") pos -= 1;

    // Even number of escapes, potential closing delimiter found
    if ((match - pos) % 2 === 1) break;

    match += 1;
  }

  // No closing delimter found.  Consume $ and continue.
  if (match === -1) {
    if (!silent) state.pending += "$";

    state.pos = start;

    return true;
  }

  // Check if we have empty content, ie: $$.  Do not parse.
  if (match - start === 0) {
    if (!silent) state.pending += "$$";

    state.pos = start + 1;

    return true;
  }

  // Check for valid closing delimiter
  res = isValidDelim(state, match);

  if (!res.canClose) {
    if (!silent) state.pending += "$";

    state.pos = start;

    return true;
  }

  if (!silent) {
    token = state.push("math_inline", "math", 0);
    token.markup = "$";
    token.content = state.src.slice(start, match);
  }

  state.pos = match + 1;

  return true;
};

const blockTex: RuleBlock = (state, start, end, silent) => {
  let firstLine;
  let lastLine;
  let next;
  let lastPos;
  let found = false;
  let pos = state.bMarks[start] + state.tShift[start];
  let max = state.eMarks[start];

  if (pos + 2 > max) return false;

  if (state.src.slice(pos, pos + 2) !== "$$") return false;

  pos += 2;
  firstLine = state.src.slice(pos, max);

  if (silent) return true;

  if (firstLine.trim().endsWith("$$")) {
    // Single line expression
    firstLine = firstLine.trim().slice(0, -2);
    found = true;
  }

  for (next = start; !found; ) {
    next += 1;

    if (next >= end) break;

    pos = state.bMarks[next] + state.tShift[next];
    max = state.eMarks[next];

    if (pos < max && state.tShift[next] < state.blkIndent)
      // non-empty line with negative indent should stop the list:
      break;

    if (state.src.slice(pos, max).trim().endsWith("$$")) {
      lastPos = state.src.slice(0, max).lastIndexOf("$$");
      lastLine = state.src.slice(pos, lastPos);
      found = true;
    }
  }

  state.line = next + 1;

  const token = state.push("math_block", "math", 0);

  token.block = true;
  token.content =
    (firstLine?.trim() ? `${firstLine}\n` : "") +
    state.getLines(start + 1, next, state.tShift[start], true) +
    (lastLine?.trim() ? lastLine : "");
  token.map = [start, state.line];
  token.markup = "$$";

  return true;
};

// set KaTeX as the renderer for markdown-it-simplemath
const katexInline = (tex: string, options: KatexOptions): string => {
  try {
    return Katex.renderToString(tex, { ...options, displayMode: false });
  } catch (error) {
    if (options.throwOnError) console.warn(error);

    return `<span class='katex-error' title='${escapeHtml(
      (error as Error).toString()
    )}'>${escapeHtml(tex)}</span>`;
  }
};

const katexBlock = (tex: string, options: KatexOptions): string => {
  try {
    return `<p class='katex-block'>${Katex.renderToString(tex, {
      ...options,
      displayMode: true,
      strict: (errorCode: string): string =>
        errorCode === "newLineInDisplayMode" ? "ignore" : "warn",
    })}</p>`;
  } catch (error) {
    if (options.throwOnError) console.warn(error);

    return `<p class='katex-block katex-error' title='${escapeHtml(
      (error as Error).toString()
    )}'>${escapeHtml(tex)}</p>`;
  }
};

export const katex: PluginWithOptions<KatexOptions> = (
  md,
  options = { throwOnError: false }
) => {
  md.inline.ruler.after("escape", "math_inline", inlineTex);
  md.block.ruler.after("blockquote", "math_block", blockTex, {
    alt: ["paragraph", "reference", "blockquote", "list"],
  });

  md.renderer.rules["math_inline"] = (tokens, index): string =>
    katexInline(tokens[index].content, options);
  md.renderer.rules["math_block"] = (tokens, index): string =>
    `${katexBlock(tokens[index].content, options)}\n`;
};
