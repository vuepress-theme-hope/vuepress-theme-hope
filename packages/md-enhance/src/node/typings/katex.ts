import type { KatexOptions as OriginalKatexOptions } from "katex";

export interface KatexOptions extends OriginalKatexOptions {
  /**
   * Whether enable mhchem extension
   *
   * 是否启用 mhchem 扩展
   *
   * @default false
   */
  mhchem?: boolean;
}
