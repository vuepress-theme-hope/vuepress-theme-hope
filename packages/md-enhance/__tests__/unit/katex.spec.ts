import { describe, it, expect, vi } from "vitest";
import MarkdownIt from "markdown-it";
import { katex } from "../../src/node/markdown-it/katex";

const markdownIt = MarkdownIt({ linkify: true }).use(katex);
const markdownItWithError = MarkdownIt({ linkify: true }).use(katex, {
  throwOnError: true,
});

describe("inline katex", () => {
  it("Shoud render", () => {
    expect(markdownIt.render(`$a=1$`)).toMatchSnapshot();
  });

  it("Should not render when escape", () => {
    expect(markdownIt.render("$a = 1\\$")).toEqual("<p>$a = 1$</p>\n");
    expect(markdownIt.render("\\$a = 1$")).toEqual("<p>$a = 1$</p>\n");
  });

  it("Should not render when having spaces", () => {
    expect(markdownIt.render(`$ a = 1 $`)).toEqual("<p>$ a = 1 $</p>\n");
  });

  it("Should not render when the ending tag is followed by number", () => {
    expect(markdownIt.render(`Of course $1 = $1`)).toEqual(
      "<p>Of course $1 = $1</p>\n"
    );
  });

  it("Should render when the first one is after a charater", () => {
    expect(markdownIt.render(`The next$a = 1$ won’t work`)).toMatchSnapshot();
  });

  it("Should not render error msg when content is wrong", () => {
    expect(markdownIt.render("$\\fra{a}{b}$")).toMatchSnapshot();
  });

  it("Should render error msg when content is wrong", () => {
    // eslint-disable-next-line @typescript-eslint/unbound-method
    const originalWarn = global.console.warn;

    global.console.warn = vi.fn();

    expect(markdownItWithError.render("$\\fra{a}{b}$")).toEqual(
      "<p><span class='katex-error' title='ParseError: KaTeX parse error: Undefined control sequence: \\fra at position 1: \\̲f̲r̲a̲{a}{b}'>\\fra{a}{b}</span></p>\n"
    );

    // eslint-disable-next-line @typescript-eslint/unbound-method
    expect(global.console.warn).toHaveBeenCalledTimes(1);
    global.console.warn = originalWarn;
  });
});

describe("block katex", () => {
  it("Shoud render", () => {
    expect(markdownIt.render(`$$a=1$$`)).toMatchSnapshot();

    expect(
      markdownIt.render(`
$$
a = 1
$$
`)
    ).toMatchSnapshot();
  });

  it("Should not render when escape", () => {
    expect(markdownIt.render("\\$\\$a = 1$$")).toEqual("<p>$$a = 1$$</p>\n");
    expect(
      markdownIt.render(`
\\$\\$
a = 1
\\$\\$
`)
    ).toEqual(`<p>$$
a = 1
$$</p>\n`);
  });

  it("Should render when having spaces", () => {
    expect(markdownIt.render(`$$ a = 1 $$`)).toMatchSnapshot();

    expect(markdownIt.render(`All $$ a = 1 $$ is true.`)).toEqual(
      "<p>All $$ a = 1 $$ is true.</p>\n"
    );
  });

  it("Should not render error msg when content is wrong", () => {
    expect(markdownIt.render("$$\\fra{a}{b}$$")).toMatchSnapshot();

    expect(
      markdownIt.render(`
$$
\\fra{a}{b}
$$
`)
    ).toMatchSnapshot();
  });

  it("Should not output warnings when content has line breaks", () => {
    expect(
      markdownIt.render(`
$$
\\begin{alignedat}{2}
    10&x+ &3&y = 2 \\\\
    3&x+&13&y = 4
\\end{alignedat}
$$    
`)
    ).toMatchSnapshot();
  });

  it("Should render error msg when content is wrong", () => {
    // eslint-disable-next-line @typescript-eslint/unbound-method
    const originalWarn = global.console.warn;

    global.console.warn = vi.fn();
    expect(markdownItWithError.render("$$\\fra{a}{b}$$")).toMatch(
      /<p class='katex-block katex-error' title='[\s\S]*?'>[\s\S]*?<\/p>/
    );

    expect(
      markdownItWithError.render(`
$$
\\fra{a}{b}
$$
`)
    ).toMatch(
      /<p class='katex-block katex-error' title='[\s\S]*?'>[\s\S]*?<\/p>/
    );

    // eslint-disable-next-line @typescript-eslint/unbound-method
    expect(global.console.warn).toHaveBeenCalledTimes(2);
    global.console.warn = originalWarn;
  });
});
