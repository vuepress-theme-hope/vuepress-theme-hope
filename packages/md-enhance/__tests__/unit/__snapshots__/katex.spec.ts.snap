// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block katex Shoud render 1`] = `
"<p class='katex-block'><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow><mi>a</mi><mo>=</mo><mn>1</mn></mrow><annotation encoding=\\"application/x-tex\\">a=1
</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.4306em;\\"></span><span class=\\"mord mathnormal\\">a</span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.6444em;\\"></span><span class=\\"mord\\">1</span></span></span></span></span></p>
"
`;

exports[`block katex Shoud render 2`] = `
"<p class='katex-block'><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow><mi>a</mi><mo>=</mo><mn>1</mn></mrow><annotation encoding=\\"application/x-tex\\">a = 1
</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.4306em;\\"></span><span class=\\"mord mathnormal\\">a</span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.6444em;\\"></span><span class=\\"mord\\">1</span></span></span></span></span></p>
"
`;

exports[`block katex Should not render error msg when content is wrong 1`] = `
"<p class='katex-block'><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow><mstyle mathcolor=\\"#cc0000\\"><mtext>\\\\fra</mtext></mstyle><mi>a</mi><mi>b</mi></mrow><annotation encoding=\\"application/x-tex\\">\\\\fra{a}{b}
</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:1em;vertical-align:-0.25em;\\"></span><span class=\\"mord text\\" style=\\"color:#cc0000;\\"><span class=\\"mord\\" style=\\"color:#cc0000;\\">\\\\fra</span></span><span class=\\"mord\\"><span class=\\"mord mathnormal\\">a</span></span><span class=\\"mord\\"><span class=\\"mord mathnormal\\">b</span></span></span></span></span></span></p>
"
`;

exports[`block katex Should not render error msg when content is wrong 2`] = `
"<p class='katex-block'><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow><mstyle mathcolor=\\"#cc0000\\"><mtext>\\\\fra</mtext></mstyle><mi>a</mi><mi>b</mi></mrow><annotation encoding=\\"application/x-tex\\">\\\\fra{a}{b}
</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:1em;vertical-align:-0.25em;\\"></span><span class=\\"mord text\\" style=\\"color:#cc0000;\\"><span class=\\"mord\\" style=\\"color:#cc0000;\\">\\\\fra</span></span><span class=\\"mord\\"><span class=\\"mord mathnormal\\">a</span></span><span class=\\"mord\\"><span class=\\"mord mathnormal\\">b</span></span></span></span></span></span></p>
"
`;

exports[`block katex Should render when having spaces 1`] = `
"<p class='katex-block'><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow><mi>a</mi><mo>=</mo><mn>1</mn></mrow><annotation encoding=\\"application/x-tex\\">a = 1 
</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.4306em;\\"></span><span class=\\"mord mathnormal\\">a</span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.6444em;\\"></span><span class=\\"mord\\">1</span></span></span></span></span></p>
"
`;

exports[`inline katex Shoud render 1`] = `
"<p><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mi>a</mi><mo>=</mo><mn>1</mn></mrow><annotation encoding=\\"application/x-tex\\">a=1</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.4306em;\\"></span><span class=\\"mord mathnormal\\">a</span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.6444em;\\"></span><span class=\\"mord\\">1</span></span></span></span></p>
"
`;

exports[`inline katex Should not render error msg when content is wrong 1`] = `
"<p><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mstyle mathcolor=\\"#cc0000\\"><mtext>\\\\fra</mtext></mstyle><mi>a</mi><mi>b</mi></mrow><annotation encoding=\\"application/x-tex\\">\\\\fra{a}{b}</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:1em;vertical-align:-0.25em;\\"></span><span class=\\"mord text\\" style=\\"color:#cc0000;\\"><span class=\\"mord\\" style=\\"color:#cc0000;\\">\\\\fra</span></span><span class=\\"mord\\"><span class=\\"mord mathnormal\\">a</span></span><span class=\\"mord\\"><span class=\\"mord mathnormal\\">b</span></span></span></span></span></p>
"
`;

exports[`inline katex Should render when the first one is after a charater 1`] = `
"<p>The next<span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mi>a</mi><mo>=</mo><mn>1</mn></mrow><annotation encoding=\\"application/x-tex\\">a = 1</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.4306em;\\"></span><span class=\\"mord mathnormal\\">a</span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.6444em;\\"></span><span class=\\"mord\\">1</span></span></span></span> won't work</p>
"
`;
